update-dois/Jenkinsfile

```diff
@@ -0,0 +1,139 @@
+						to: '$DEFAULT_RECIPIENTS',
+						subject: "UpdateDOIs List for v${env.RELEASE_NUMBER}",
+						attachmentsPattern: "**/doisToBeUpdated-v${env.RELEASE_NUMBER}.txt"
+					)
+				}
+			}
+		}
+		// UpdateDOIs should pause at this stage until the curator confirms the report file is correct. Once they do, respond with 'yes' to the user input form that Jenkins brings up.
+		stage('User Input Required: Confirm DOIs'){
+			steps{
+				script{
+					// This brings up a user input form, asking for confirmation that the curator overseeing release approves of the report file they received.
+					def userInput = input(
+						id: 'userInput', message: "Has a curator confirmed that the list of DOIs to be updated in doisToBeUpdated-v${env.RELEASE_NUMBER}.txt is correct? (yes/no)",
+						parameters: [
+							[$class: 'TextParameterDefinition', defaultValue: '', description: 'Confirmation of updateable DOIs', name: 'response']
+						])
+
+					if (userInput.toLowerCase().startsWith("y")) {
+						echo("Proceeding to UpdateDOIs step.")
+					} else {
+						error("Please confirm output of UpdateDOIs Test Run matches DOIs that should be updated")
+					}
+				}
+			}
+		}
+		// Now that you have curator approval regarding the report file, this step executes the same jar file again -- this time providing the report file as an argument.
+		// With the report file as the second argument, UpdateDOIs executes database modifications.
+		stage('Main: UpdateDOIs'){
+			steps{
+				script{
+					dir('update-dois'){
+						withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]) {
+							sh "java -jar target/update-dois-${env.UPDATE_DOIS_VERSION}-jar-with-dependencies.jar $ConfigFile doisToBeUpdated-v${env.RELEASE_NUMBER}.txt"
+						}
+					}
+				}
+			}
+		}
+		// This stage backs up the gk_central and release_current databases after they are modified.
+		stage('Post: Backup DBs'){
+			steps{
+				script{
+					dir('update-dois'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							def release_current_after_update_dois_dump = "${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_after_update_dois.dump"
+							def central_before_after_dois_dump = "${env.GK_CENTRAL}_${env.RELEASE_NUMBER}_after_update_dois.dump"
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_after_update_dois_dump"
+							sh "gzip -f $release_current_after_update_dois_dump"
+							sh "mysqldump -u$user -p$pass -h${env.CURATOR_SERVER} ${env.GK_CENTRAL} > $central_before_after_dois_dump"
+							sh "gzip -f $central_before_after_dois_dump"
+						}
+					}
+				}
+			}
+		}
+		// This stage archives all logs and database backups produced by UpdateDOIs.
+		stage('Archive logs and backups'){
```

Comment:
Also: Why are backing up the database and archive the logs and backup files two separate stages? To me it would make sense to group them as two steps of a single "Post-UpdateDOI" stage. Is there an advantage to breaking everything up such that each stage only contains one step?

Reply:
No there isn't a particular advantage, aside from allowing for easier identification of failures (but that is sort of trivial between DB backup and archiving). I was just trying to be explicit for stage naming/theming. I can have a single 'post' step that includes backups and archiving if wanted

Reply:
Being able to easily identify failures is a Good Thing, so in that case, maybe it's better to have them separate.